test/unit/version_test.rb

```diff
@@ -730,33 +748,37 @@ class VersionTest < ActiveSupport::TestCase
       assert_equal @spec.required_rubygems_version.to_s,  @version.required_rubygems_version
     end
 
-    context "metadata" do
+    context "with metadata" do
       should "be invalid with empty string as link" do
-        assert_raise ActiveRecord::RecordInvalid do
-          @spec.metadata = { "home" => "" }
-          @version.update_attributes_from_gem_specification!(@spec)
-        end
+        @version.metadata = { "home" => "" }
+        @version.validate
+        assert_equal @version.errors.messages[:metadata], ["['home'] does not appear to be a valid URL"]
       end
 
       should "be invalid with invalid link" do
-        assert_raise ActiveRecord::RecordInvalid do
-          @spec.metadata = { "home" => "http:/github.com/bestgemever" }
-          @version.update_attributes_from_gem_specification!(@spec)
-        end
+        @version.metadata = { "home" => "http:/github.com/bestgemever" }
+        @version.validate
+        assert_equal @version.errors.messages[:metadata], ["['home'] does not appear to be a valid URL"]
       end
 
       should "be valid with valid link" do
-        assert_nothing_raised do
-          @spec.metadata = { "home" => "http://github.com/bestgemever" }
-          @version.update_attributes_from_gem_specification!(@spec)
-        end
+        @version.metadata = { "home" => "http://github.com/bestgemever" }
+        assert @version.validate
+        assert_equal @version.errors.messages[:metadata], []
       end
 
-      should "be invalid with maxiumum length" do
-        assert_raise ActiveRecord::RecordInvalid do
-          @spec.metadata = { "home" => ("hello" * 2000) }
-          @version.update_attributes_from_gem_specification!(@spec)
-        end
```

Comment:
changed this to check for error message explicitly as existing test was raising invalid record for home url being invalid (not max length).